### ISSUE TITLE
Separate Site and Controller repositories

### ISSUE THREAD

**Issue by squeedee** [vmware-tanzu/cartographer#436]
### Description of problem

1. Site content changes at a different rate than the code
2. We often need to update versioned docs separate from the tag for that version (site is always built from a single branch, namely `main`)
3. Waiting for code checks on PRs when the only thing we've changed is documentation, is a significant pain
4. If we do end up creating a multi-product site (Eg, introduce runnable as it's own project) then we'll be out ahead of the infra changes needed (namely, seperating the Site repo :D)

### Plan:

* [X] create cartographer-site repo  [DONE](https://github.com/vmware-tanzu/cartographer-site)
* [x] Place all migration work on a branch [initial-migration](https://github.com/vmware-tanzu/cartographer-site/tree/initial-migration)
  * [x] ensure all branches for site work is merged
  * [x] Move experimental editor over and patch the release code
  * [x] Deal with the CRD generation code so it bases itself off the latest release in the cartographer repository
  * [x] call for a moratorium on site PRs (explain why) in Slack, and put a [Blocked on Migration] tag on all related doc stories.
  * [x] Ensure Linter runs in github actions
  * [x] Ensure dependabot is configured
  * [x] Configure require PR approval
* [x] Make PR for initial-migration and get it merged
* prepare cartographer PR:
  * [x] Clean up any call-through makefile targets in the cartographer makefile (they will no longer be able to reach the site) (dont merge!)
  * [x] remove site and experimental live-editor from cartographer repo (Don't merge!)
* [x] Have netlify use the new repository
* [x] Merge cartographer PR

* [x] unblock stories and let Slack know that the migration is complete

**Comment by scothis** [comment]
An intermediate option between same branch and different repos is to move website publishing to the gh-pages branch. It does mean that PRs need to be more considerate about which branch is the base.

**Comment by squeedee** [comment]
> An intermediate option between same branch and different repos is to move website publishing to the gh-pages branch. It does mean that PRs need to be more considerate about which branch is the base.

Do you propose this because of the pain of obtaining another repository? Otherwise it's pretty slight for me - in fact I'd prefer the separation of concerns and the effort involved over the weirdness of a branch.

Would love for these steps to also help ease the plan, up front, for adopting other subsystems.

**Comment by scothis** [comment]
> Do you propose this because of the pain of obtaining another repository?

Nope, getting a new repo should be easy. Multi-repo vs mono-repo is always a balancing act

**Comment by squeedee** [comment]
My feeling is that adding conditionals to GH actions to separate PR testing
is ugly and a maintenance concern. So is the aforementioned PR targets.
When we (I think inevitably) have maintenance branches for the binary,
updating `site/docs/v#.#.#` on one branch to match changes to the branch/
`v#.#.#-main` (or even `main`) seems more cumbersome than 2 repos. I
realize people can practice taking two checkouts, but still unwieldy IMHO.
This feels like the right degree of separation, but happy to hear
counterpoint.

**Comment by scothis** [comment]
There are no conditionals required for actions. Since the workflows are defined in source, you can have different workflows in different branches for pushes/prs.

**Comment by squeedee** [comment]
Which implies an intentionally different structure of the content of the branch (I'd rip out all the code) at which point, what benefit the branch over a new repo with clear concerns?

**Comment by martyspiewak** [comment]
I moved the last task for removing patch versions from the docs dropdown to a new issue in the cartographer-site repo: https://github.com/vmware-tanzu/cartographer-site/issues/20

Closing this as It looks like the rest of the tasks have been completed.
